src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/ServiceToCcdServiceProcessor.java

```diff
@@ -0,0 +1,88 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static java.util.stream.Stream.of;
+
+@Slf4j
+@Component
+public class ServiceToCcdServiceProcessor extends JsrValidationBaseProcessor<ServiceToCcdService> {
+
+    @Autowired
+    JsrValidatorInitializer<ServiceToCcdService> serviceToCcdServiceJsrValidatorInitializer;
+
+    @Value("${logging-component-name}")
+    private String logComponentName;
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void process(Exchange exchange) throws Exception {
+
+        List<ServiceToCcdService> serviceToCcdServices;
+
+        serviceToCcdServices = exchange.getIn().getBody() instanceof List
+            ? (List<ServiceToCcdService>) exchange.getIn().getBody()
+            : singletonList((ServiceToCcdService) exchange.getIn().getBody());
+
+        log.info(" {} ServiceToCCDService Records count before Validation & before merging service names {}::",
+                 logComponentName, serviceToCcdServices.size()
+        );
+
+        List<ServiceToCcdService> refinedServiceToCcdServices = populateService(serviceToCcdServices);
+        log.info(" {} ServiceToCCDService Records count before Validation & after merging service names {}::",
+                 logComponentName, refinedServiceToCcdServices.size()
+        );
+
+        List<ServiceToCcdService> filteredServiceToCcdServices = validate(
+            serviceToCcdServiceJsrValidatorInitializer,
+            refinedServiceToCcdServices
+        );
+        log.info(" {} ServiceToCCDService Records count after Validation & after merging service names {}::",
+                 logComponentName, filteredServiceToCcdServices.size()
+        );
+
+        audit(serviceToCcdServiceJsrValidatorInitializer, exchange);
+
+        if (filteredServiceToCcdServices.size() == 0) {
+            log.error(" {} ServiceToCcdService failed as no valid records present::", logComponentName);
+
+            throw new RouteFailedException("ServiceToCcdService failed as no valid records present");
+        }
+        exchange.getMessage().setBody(filteredServiceToCcdServices);
+    }
+
+    /**
+     * Spits list by , serviceNames and recreates the list.
+     *
+     * @param serviceToCcdServices serviceToCcdServices
+     * @return List
+     */
+    private List<ServiceToCcdService> populateService(List<ServiceToCcdService> serviceToCcdServices) {
+        List<ServiceToCcdService> refinedServiceToCcdServices = new ArrayList<>();
+        serviceToCcdServices.forEach(serviceToCcdService ->
+                                         of(serviceToCcdService.getCcdServiceName().split(","))
+                                             .forEach(serviceNames -> {
+                                                 refinedServiceToCcdServices.add(ServiceToCcdService.builder()
+                                                                                     .serviceCode(serviceToCcdService
```

Comment:
this logic will change based on the column name comment mentioned on the file.
Column which can have comma separated values is CCD CASE TYPE

Reply:
done